### PR TITLE
[SL-2420] Panel Edit Metadata + Create Object Modal

### PR DESCRIPTION
**Pull request by james-wallis**
- Adds functionality to make Metadata editable, editing a field will switch the Panel into edit mode
- Also adds a Create Modal to create objects

Also:
- Fixes the Panel on mobile from opening too large
- Adds support for Availability by supporting the new `Visible` Interface. When it isn't available it defaults to `Metadata` to support Beta 1 envs
- 

* [x] Feature

#### Jira

https://skylarkplatform.atlassian.net/browse/SL-2420
https://skylarkplatform.atlassian.net/browse/SL-2415

### CODE REVIEW

File: src/components/panel/panelInputs/panelInputs.component.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import {
+  DetailedHTMLProps,
+  HTMLInputTypeAttribute,
+  InputHTMLAttributes,
+} from "react";
+
+import { NormalizedObjectFieldType } from "src/interfaces/skylark";
+
+export const convertFieldTypeToInputType = (
```
Comment (james-wallis) [inline src/components/panel/panelInputs/panelInputs.component.tsx:9]: ToHTMLInputType

File: src/components/inputs/select/languageSelect/languageSelect.component.tsx [hunk 2]
```diff
@@ -22,9 +26,9 @@ export const LanguageSelect = ({
     options={
       languages?.map((lang) => ({ value: lang, label: lang })) || options
     }
-    className={clsx(props.variant === "pill" ? "w-20" : "w-36")}
+    className={clsx(props.variant === "pill" ? "w-20" : props.className)}
     placeholder="Language"
-    rounded
+    rounded={props.rounded === undefined ? true : props.rounded}
```
Comment (freixedelo) [inline src/components/inputs/select/languageSelect/languageSelect.component.tsx:31]: we could do `props.rounded || true` ?

File: src/components/inputs/select/select.component.tsx [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+  currentSelected?: SelectOption;
+  style?: CSSProperties;
+}) => (
+  <Combobox.Option
+    key={option.value}
+    className={({ active }) =>
+      clsx(
+        "relative flex cursor-default select-none items-center text-gray-900",
+        variant === "pill" ? "px-2" : "px-4 pl-6",
+        (active || currentSelected?.value === option.value) &&
```
Comment (freixedelo) [inline src/components/inputs/select/select.component.tsx:56]: is `active` different from `currentSelected?.value === option.value` ?

File: src/components/inputs/select/select.component.tsx [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+          "bg-ultramarine-50",
+      )
+    }
+    value={option}
+    style={style}
+  >
+    <span
+      className={clsx(
+        "block truncate",
+        currentSelected?.value === option.value ? "font-medium" : "font-normal",
```
Comment (freixedelo) [inline src/components/inputs/select/select.component.tsx:66]: is so, don't we need `active` here ?

File: src/components/inputs/skylarkObjectFieldInput/skylarkObjectFieldInput.component.tsx [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+  <textarea
+    {...register(field, registerOptions)}
+    aria-invalid={error ? "true" : "false"}
+    id={createHtmlForId(field)}
+    rows={
+      (value &&
+        (((value as string).length > 1000 && 18) ||
+          ((value as string).length > 150 && 9) ||
+          ((value as string).length > 50 && 5))) ||
+      1
```
Comment (freixedelo) [inline src/components/inputs/skylarkObjectFieldInput/skylarkObjectFieldInput.component.tsx:121]: Do we want a `textarea` with one row? that is an `input`

File: src/components/inputs/skylarkObjectFieldInput/skylarkObjectFieldInput.component.tsx [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+    error,
+    registerOptions: {
+      required,
+      validate: (value) => {
+        try {
+          if (
+            (config.type === "int" || config.type === "float") &&
+            value !== ""
+          ) {
+            if (
```
Comment (freixedelo) [inline src/components/inputs/skylarkObjectFieldInput/skylarkObjectFieldInput.component.tsx:193]: not sure but seems that we could merge this `if` with the previous one 🤔

File: src/lib/skylark/objects.ts [hunk 7]
```diff
@@ -177,9 +189,10 @@ export const getObjectOperations = (
   const { relationships, ...createMeta } = getMutationInfo(
     createMutation,
     objectType,
-    "create",
   );
-  const updateMeta = getMutationInfo(updateMutation, objectType, "update");
+  const updateMeta = getMutationInfo(updateMutation, objectType);
+
+  const hasRelationships = relationships.length > 0;
```
Comment (freixedelo) [inline src/lib/skylark/objects.ts:195]: not sure if this is enough 🤔 if not, it'll be fixed on merge

Comment (freixedelo) [reply src/lib/skylark/objects.ts:195]: enough to make sure that has relationships 😄

**Review by freixedelo — APPROVED** [review]
excellent work

File: src/lib/skylark/objects.ts (hunk 7 above)
Comment (james-wallis) [reply src/lib/skylark/objects.ts:195]: Niceee

File: src/components/inputs/skylarkObjectFieldInput/skylarkObjectFieldInput.component.tsx (hunk 5 above)
Comment (james-wallis) [reply src/components/inputs/skylarkObjectFieldInput/skylarkObjectFieldInput.component.tsx:121]: So they can add more content and expand it

File: src/components/inputs/skylarkObjectFieldInput/skylarkObjectFieldInput.component.tsx [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+const SkylarkObjectFieldInputGeneric = ({
+  field,
+  register,
+  config,
+  error,
+  registerOptions,
+}: SkylarkObjectFieldInputComponentProps) => (
+  <input
+    {...register(field, {
+      // valueAsNumber: (config.type === "int" || config.type === "float") as
```
[the remark this answers is not in the document]
Comment (james-wallis) [reply src/components/inputs/skylarkObjectFieldInput/skylarkObjectFieldInput.component.tsx]: nop

File: src/components/inputs/select/languageSelect/languageSelect.component.tsx (hunk 2 above)
Comment (james-wallis) [reply src/components/inputs/select/languageSelect/languageSelect.component.tsx:31]: that wouldn't handle `props.rounded = false`

File: src/components/inputs/select/select.component.tsx (hunk 4 above)
Comment (james-wallis) [reply src/components/inputs/select/select.component.tsx:66]: I think active is like hovered over

**Pull request merged**